src/transformers/models/gemma/modeling_gemma.py

```diff
@@ -276,7 +276,7 @@ def forward(
 
         attn_output = attn_output.transpose(1, 2).contiguous()
 
-        attn_output = attn_output.reshape(bsz, q_len, self.hidden_size)
+        attn_output = attn_output.view(bsz, q_len, -1)
```

Comment:
This is the only modelling code change required - the remainder of the changes in this PR are logit + integration tests